ethcore/src/client/client.rs

```diff
@@ -259,12 +259,12 @@ impl Client {
 		// could be retracted in import `k+1`. This is why to understand if after all inserts
 		// the block is enacted or retracted we iterate over all routes and at the end final state
 		// will be in the hashmap
-		let map = import_results.into_iter().fold(HashMap::new(), |mut map, route| {
-			for hash in route.enacted {
-				map.insert(hash, true);
+		let map = import_results.iter().fold(HashMap::new(), |mut map, route| {
+			for hash in &route.enacted {
+				map.insert(hash.clone(), true);
 			}
-			for hash in route.retracted {
-				map.insert(hash, false);
+			for hash in &route.retracted {
+				map.insert(hash.clone(), false);
```

Comment:
could drastically simplify this function as

```rust
let (mut enacted, mut retracted) = (Vec::new(), Vec::new());
for route in import_results {
    enacted.extend(route.enacted.iter().cloned());
    retracted.extend(route.retracted.iter().cloned());
}

(enacted, retracted)
```

simpler, fewer allocations, shorter